pkg/types/azure/validation/platform.go

```diff
@@ -0,0 +1,41 @@
+package validation
+
+import (
+	"sort"
+
+	"github.com/openshift/installer/pkg/types/azure"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	// Regions is a map of the known azure regions. The key of the map is
+	// the short name of the region. The value of the map is the long
+	// name of the region.
+	Regions = map[string]string{
+		"eastus": "East US",
+		"westus": "West US",
+	}
+
+	validRegionValues = func() []string {
+		validValues := make([]string, len(Regions))
+		i := 0
+		for r := range Regions {
+			validValues[i] = r
+			i++
+		}
+		sort.Strings(validValues)
+		return validValues
+	}()
+)
+
+// ValidatePlatform checks that the specified platform is valid.
+func ValidatePlatform(p *azure.Platform, fldPath *field.Path) field.ErrorList {
```

Comment:
this is implemented but not called, we'll need to add it to `validatePlatform` here https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L192-L221

Reply:
Also these error messages need to be updated:
```
    --- FAIL: TestValidateInstallConfig/missing_platform (0.00s)
    	installconfig_test.go:441: 
    			Error Trace:	installconfig_test.go:441
    			Error:      	Expect "platform: Invalid value: types.Platform{AWS:(*aws.Platform)(nil), Libvirt:(*libvirt.Platform)(nil), None:(*none.Platform)(nil), OpenStack:(*openstack.Platform)(nil), Azure:(*azure.Platform)(nil)}: must specify one of the platforms (aws, azure, none, openstack)" to match "^platform: Invalid value: types\.Platform{AWS:\(\*aws\.Platform\)\(nil\), Libvirt:\(\*libvirt\.Platform\)\(nil\), None:\(\*none\.Platform\)\(nil\), OpenStack:\(\*openstack\.Platform\)\(nil\)}: must specify one of the platforms \(aws, none, openstack\)$"
    			Test:       	TestValidateInstallConfig/missing_platform
```

Reply:
both were addressed